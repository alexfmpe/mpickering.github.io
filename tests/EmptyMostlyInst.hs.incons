[(tests/examples/EmptyMostlyInst.hs:7:9-14,
  (AnnVal, [tests/examples/EmptyMostlyInst.hs:7:11-12]))]