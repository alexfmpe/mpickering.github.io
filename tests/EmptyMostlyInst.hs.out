{-# LANGUAGE FlexibleInstances #-}
module EmptyMostlyInst where
   {
;;;;;;;;;;;;
;
instance Eq (Int,Integer) where {;;;;;;;;;
;;;;;;; a == b = False;;;;;;;;;;;
}
   }

==============
tests/examples/EmptyMostlyInst.hs
==============
lengths:(168,168)

==============

({ tests/examples/EmptyMostlyInst.hs:1:1 }
 Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((AnnComment DComment (DP (0,0),DP (0,34)) "{-# LANGUAGE FlexibleInstances #-}"),DP (0,0)),((G AnnModule),DP (1,1)),((G AnnVal),DP (0,1)),((G AnnWhere),DP (0,1)),((G AnnOpenC),DP (1,4)),((G AnnSemi),DP (1,1)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (1,1)),((G AnnCloseC),DP (1,4)),((G AnnEofPos),DP (1,0))])
 (HsModule 
  (Just 
   ({ tests/examples/EmptyMostlyInst.hs:2:8-22 }
    Nothing{ModuleName: EmptyMostlyInst})) 
  (Nothing) 
  [] 
  [
   ({ tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1) }
    Just (Ann (DP (1,1)) (ColDelta 1) DP (1,1) [] [((G AnnInstance),DP (0,0)),((G AnnWhere),DP (0,1)),((G AnnOpenC),DP (0,1)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (1,1)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnCloseC),DP (1,1))])
    (InstD 
     (ClsInstD 
      (ClsInstDecl 
       ({ tests/examples/EmptyMostlyInst.hs:6:10-25 }
        Just (Ann (DP (0,1)) (ColDelta 10) DP (0,1) [] [])
        (HsForAllTy 
         (Implicit) 
         (Nothing) 
         (HsQTvs 
          [] 
          []) 
         ({ <no location info> }
          Nothing
          []) 
         ({ tests/examples/EmptyMostlyInst.hs:6:10-25 }
          Just (Ann (DP (0,0)) (ColDelta 10) DP (0,0) [] [])
          (HsAppTy 
           ({ tests/examples/EmptyMostlyInst.hs:6:10-11 }
            Just (Ann (DP (0,0)) (ColDelta 10) DP (0,0) [] [((G AnnVal),DP (0,0))])
            (HsTyVar 
             (Unqual {OccName: Eq}))) 
           ({ tests/examples/EmptyMostlyInst.hs:6:13-25 }
            Just (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0))])
            (HsTupleTy 
             (HsBoxedOrConstraintTuple) 
             [
              ({ tests/examples/EmptyMostlyInst.hs:6:14-16 }
               Just (Ann (DP (0,0)) (ColDelta 14) DP (0,0) [] [((G AnnVal),DP (0,0)),((G AnnComma),DP (0,0))])
               (HsTyVar 
                (Unqual {OccName: Int}))),
              ({ tests/examples/EmptyMostlyInst.hs:6:18-24 }
               Just (Ann (DP (0,0)) (ColDelta 18) DP (0,0) [] [((G AnnVal),DP (0,0))])
               (HsTyVar 
                (Unqual {OccName: Integer})))])))))) {Bag(Located (HsBind RdrName)): 
       [
        ({ tests/examples/EmptyMostlyInst.hs:7:9-22 }
         Just (Ann (DP (0,1)) (ColDelta 9) DP (0,1) [] [])
         (FunBind 
          ({ tests/examples/EmptyMostlyInst.hs:7:11-12 }
           Just (Ann (DP (0,1)) (ColDelta 11) DP (0,1) [] [((G AnnVal),DP (0,0))])
           (Unqual {OccName: ==})) 
          (True) 
          (MG 
           [
            ({ tests/examples/EmptyMostlyInst.hs:7:9-22 }
             Just (Ann (DP (0,0)) (ColDelta 9) DP (0,0) [] [((G AnnEqual),DP (0,1)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0))])
             (Match 
              (Just 
               ((,) 
                ({ tests/examples/EmptyMostlyInst.hs:7:11-12 }
                 Just (Ann (DP (0,1)) (ColDelta 11) DP (0,1) [] [((G AnnVal),DP (0,0))])
                 (Unqual {OccName: ==})) 
                (True))) 
              [
               ({ tests/examples/EmptyMostlyInst.hs:7:9 }
                Just (Ann (DP (0,0)) (ColDelta 9) DP (0,0) [] [((G AnnVal),DP (0,0))])
                (VarPat 
                 (Unqual {OccName: a}))),
               ({ tests/examples/EmptyMostlyInst.hs:7:14 }
                Just (Ann (DP (0,1)) (ColDelta 14) DP (0,1) [] [((G AnnVal),DP (0,0))])
                (VarPat 
                 (Unqual {OccName: b})))] 
              (Nothing) 
              (GRHSs 
               [
                ({ tests/examples/EmptyMostlyInst.hs:7:16-22 }
                 Just (Ann (DP (0,-1)) (ColDelta 16) DP (0,-1) [] [])
                 (GRHS 
                  [] 
                  ({ tests/examples/EmptyMostlyInst.hs:7:18-22 }
                   Just (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])
                   (HsVar 
                    (Unqual {OccName: False})))))] 
               (EmptyLocalBinds))))] 
           [] 
           (PlaceHolder) 
           (FromSource)) 
          (WpHole) 
          (PlaceHolder) 
          []))]} 
       [] 
       [] 
       [] 
       (Nothing)))))] 
  (Nothing) 
  (Nothing)))
==============
[(AnnKey tests/examples/EmptyMostlyInst.hs:1:1 CN "HsModule",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((AnnComment DComment (DP (0,0),DP (0,34)) "{-# LANGUAGE FlexibleInstances #-}"),DP (0,0)),((G AnnModule),DP (1,1)),((G AnnVal),DP (0,1)),((G AnnWhere),DP (0,1)),((G AnnOpenC),DP (1,4)),((G AnnSemi),DP (1,1)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (1,1)),((G AnnCloseC),DP (1,4)),((G AnnEofPos),DP (1,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1) CN "InstD",
  (Ann (DP (1,1)) (ColDelta 1) DP (1,1) [] [((G AnnInstance),DP (0,0)),((G AnnWhere),DP (0,1)),((G AnnOpenC),DP (0,1)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (1,1)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnSemi),DP (0,0)),((G AnnCloseC),DP (1,1))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:6:10-11 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 10) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:6:10-25 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 10) DP (0,0) [] [])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:6:10-25 CN "HsForAllTy",
  (Ann (DP (0,1)) (ColDelta 10) DP (0,1) [] [])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:6:13-25 CN "HsTupleTy",
  (Ann (DP (0,1)) (ColDelta 13) DP (0,1) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:6:14-16 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 14) DP (0,0) [] [((G AnnVal),DP (0,0)),((G AnnComma),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:6:18-24 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 18) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:9 CN "VarPat",
  (Ann (DP (0,0)) (ColDelta 9) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:9-22 CN "FunBind",
  (Ann (DP (0,1)) (ColDelta 9) DP (0,1) [] [])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:9-22 CN "Match",
  (Ann (DP (0,0)) (ColDelta 9) DP (0,0) [] [((G AnnEqual),DP (0,1)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0)),(AnnSemiSep,DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:11-12 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 11) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:14 CN "VarPat",
  (Ann (DP (0,1)) (ColDelta 14) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:16-22 CN "GRHS",
  (Ann (DP (0,-1)) (ColDelta 16) DP (0,-1) [] [])),
 (AnnKey tests/examples/EmptyMostlyInst.hs:7:18-22 CN "HsVar",
  (Ann (DP (0,1)) (ColDelta 18) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey <no location info> CN "EmptyLocalBinds",
  (Ann (DP (-7,0)) (ColDelta 0) DP (-7,0) [] []))]
==============
([((tests/examples/EmptyMostlyInst.hs:1:1, AnnCloseC),
   [tests/examples/EmptyMostlyInst.hs:9:4]),
  ((tests/examples/EmptyMostlyInst.hs:1:1, AnnModule),
   [tests/examples/EmptyMostlyInst.hs:2:1-6]),
  ((tests/examples/EmptyMostlyInst.hs:1:1, AnnOpenC),
   [tests/examples/EmptyMostlyInst.hs:3:4]),
  ((tests/examples/EmptyMostlyInst.hs:1:1, AnnSemi),
   [tests/examples/EmptyMostlyInst.hs:5:1,
    tests/examples/EmptyMostlyInst.hs:4:12,
    tests/examples/EmptyMostlyInst.hs:4:11,
    tests/examples/EmptyMostlyInst.hs:4:10,
    tests/examples/EmptyMostlyInst.hs:4:9,
    tests/examples/EmptyMostlyInst.hs:4:8,
    tests/examples/EmptyMostlyInst.hs:4:7,
    tests/examples/EmptyMostlyInst.hs:4:6,
    tests/examples/EmptyMostlyInst.hs:4:5,
    tests/examples/EmptyMostlyInst.hs:4:4,
    tests/examples/EmptyMostlyInst.hs:4:3,
    tests/examples/EmptyMostlyInst.hs:4:2,
    tests/examples/EmptyMostlyInst.hs:4:1]),
  ((tests/examples/EmptyMostlyInst.hs:1:1, AnnWhere),
   [tests/examples/EmptyMostlyInst.hs:2:24-28]),
  ((tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1), AnnCloseC),
   [tests/examples/EmptyMostlyInst.hs:8:1]),
  ((tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1), AnnInstance),
   [tests/examples/EmptyMostlyInst.hs:6:1-8]),
  ((tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1), AnnOpenC),
   [tests/examples/EmptyMostlyInst.hs:6:33]),
  ((tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1), AnnSemi),
   [tests/examples/EmptyMostlyInst.hs:7:7,
    tests/examples/EmptyMostlyInst.hs:7:6,
    tests/examples/EmptyMostlyInst.hs:7:5,
    tests/examples/EmptyMostlyInst.hs:7:4,
    tests/examples/EmptyMostlyInst.hs:7:3,
    tests/examples/EmptyMostlyInst.hs:7:2,
    tests/examples/EmptyMostlyInst.hs:7:1,
    tests/examples/EmptyMostlyInst.hs:6:42,
    tests/examples/EmptyMostlyInst.hs:6:41,
    tests/examples/EmptyMostlyInst.hs:6:40,
    tests/examples/EmptyMostlyInst.hs:6:39,
    tests/examples/EmptyMostlyInst.hs:6:38,
    tests/examples/EmptyMostlyInst.hs:6:37,
    tests/examples/EmptyMostlyInst.hs:6:36,
    tests/examples/EmptyMostlyInst.hs:6:35,
    tests/examples/EmptyMostlyInst.hs:6:34]),
  ((tests/examples/EmptyMostlyInst.hs:(6,1)-(8,1), AnnWhere),
   [tests/examples/EmptyMostlyInst.hs:6:27-31]),
  ((tests/examples/EmptyMostlyInst.hs:6:13-25, AnnCloseP),
   [tests/examples/EmptyMostlyInst.hs:6:25]),
  ((tests/examples/EmptyMostlyInst.hs:6:13-25, AnnOpenP),
   [tests/examples/EmptyMostlyInst.hs:6:13]),
  ((tests/examples/EmptyMostlyInst.hs:6:14-16, AnnComma),
   [tests/examples/EmptyMostlyInst.hs:6:17]),
  ((tests/examples/EmptyMostlyInst.hs:7:9-14, AnnVal),
   [tests/examples/EmptyMostlyInst.hs:7:11-12]),
  ((tests/examples/EmptyMostlyInst.hs:7:9-22, AnnEqual),
   [tests/examples/EmptyMostlyInst.hs:7:16]),
  ((tests/examples/EmptyMostlyInst.hs:7:9-22, AnnFunId),
   [tests/examples/EmptyMostlyInst.hs:7:11-12]),
  ((tests/examples/EmptyMostlyInst.hs:7:9-22, AnnSemi),
   [tests/examples/EmptyMostlyInst.hs:7:23,
    tests/examples/EmptyMostlyInst.hs:7:24,
    tests/examples/EmptyMostlyInst.hs:7:25,
    tests/examples/EmptyMostlyInst.hs:7:26,
    tests/examples/EmptyMostlyInst.hs:7:27,
    tests/examples/EmptyMostlyInst.hs:7:28,
    tests/examples/EmptyMostlyInst.hs:7:29,
    tests/examples/EmptyMostlyInst.hs:7:30,
    tests/examples/EmptyMostlyInst.hs:7:31,
    tests/examples/EmptyMostlyInst.hs:7:32,
    tests/examples/EmptyMostlyInst.hs:7:33]),
  ((<no location info>, AnnEofPos),
   [tests/examples/EmptyMostlyInst.hs:10:1])],
 [(<no location info>,
   [AnnBlockComment "{-# LANGUAGE FlexibleInstances #-}"])])